### PR TITLE
Add the dump reader

### PR DESCRIPTION
**Pull request by Ladsgroup**
I tested some parts but it would be good if we test it more.

### CODE REVIEW

File: editquality/utilities/extract_balanced_sample.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    revert_radius = int(args['--revert-radius'])
+    revert_window = int(args['--revert-window'])
+
+    if args['--host']:
+        session = mwapi.Session(
+            args['--host'], user_agent="ORES revert labeling utility")
+    else:
+        session = None
+    dump = bz2.BZ2File(args['--dump'], 'r')
+    dump = mwxml.Dump.from_file(dump)
```
Comment (halfak) [inline editquality/utilities/extract_balanced_sample.py]: See `mwxml.map`.  We can operate on many XML files at once.

**Comment by he7d3r** [comment]
Didn't test, but looks good.

File: editquality/utilities/extract_balanced_sample.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        verbose=verbose)
+
+
+def run(dumps, session, start, end, revert_radius, revert_window,
+        reverted_only, trusted_groups, min_editcount, rev_reverteds,
+        verbose=False):
+
+    user_data = {}
+    for page in mwxml.map(page_data, dumps):
+        detector = mwreverts.Detector()
```
Comment (halfak) [inline editquality/utilities/extract_balanced_sample.py]: Did you test this?  It looks like it won't work.  You need to iterate through the page in the dump_processor function.

**Comment by halfak** [comment]
All looks good to me.  I'm just running a test with it against English Wikipedia now.

File: editquality/utilities/extract_damaging.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    <dump-file>                 Path to dump file.
+    --host=<url>                The host URL of the MediaWiki install where an
+                                API can be found.
+    --start=<timestamp>         Start time.
+    --end=<timestamp>           End time.
+    --reverted-only             Only mark reverted edits as potentially
+                                damaging
+    --revert-radius=<revs>      The maximum amount of revisions that a
+                                reverting edit can revert [default: 15]
+    --revert-window=<secs>      The maximum amount of time to wait for a
```
Comment (halfak) [inline editquality/utilities/extract_damaging.py]: We should probably let the user provide this in hours and do the conversion in code.

File: editquality/utilities/extract_damaging.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+                                revision to be reverted [default: 172800]
+    --trusted-groups=<groups>   User groups that should be considered trusted.
```
Comment (halfak) [inline editquality/utilities/extract_damaging.py:34]: Should note that comma delimited syntax is expected.

**Comment by Ladsgroup** [comment]
Done :)

**Comment by halfak** [comment]
I submitted a few fixes and did a test run.  Looks good!

**Pull request merged**